Basic/DistributedAuthoritySocialHub/Assets/Scripts/Services/VivoxManager.cs

```diff
@@ -9,14 +11,29 @@ namespace Unity.Multiplayer.Samples.SocialHub.Services
 {
     class VivoxManager : MonoBehaviour
     {
-        internal static VivoxManager Instance { get; set; }
-        string ChannelName { get; set; }
+        string m_TextChannelName;
+        string m_VoiceChannelName;
+
+        internal static VivoxManager Instance { get; private set; }
+
+        PlayersTopUIController m_PlayersTopUIController;
+        PlayersTopUIController PlayersTopUIController
+        {
+            get
+            {
+                if (m_PlayersTopUIController == null)
+                    m_PlayersTopUIController = FindFirstObjectByType<PlayersTopUIController>();
+
+                return m_PlayersTopUIController;
+            }
+        }
 
         void Awake()
         {
             if (Instance == null)
             {
                 Instance = this;
+                StartCoroutine(RequestMicrophonePermissions());
```

Comment:
Could this be prompted later on in the application if the user declines initially?

Reply:
After testing on Windows, I didn't receive a prompt for permissions. I should have right?

Reply:
Microphone was working without this on Windows, just on Mac and iOS the permission has to be requested.
https://docs.unity3d.com/6000.0/Documentation/ScriptReference/Application.RequestUserAuthorization.html

I did not check for platform to make this request as it had no side-effect on not apple platforms. Changed the Signature to RequestMicrophonePermissionsIOsMacOS

Reply:
Gotcha.
What do you think about wrapping the method too in defines?

And interesting question. Perhaps we could wrap the request for permission whenever the user tries to modify volume or unmute?

Reply:
I think it is fine to ask once, i think that's what most apps do